internal/impl/kafka/enterprise/topic_crud.go

```diff
@@ -56,7 +57,38 @@ func createTopic(ctx context.Context, topic string, replicationFactorOverride bo
 		}
 	}
 
-	if _, err := outputAdminClient.CreateTopic(ctx, partitions, rp, nil, topic); err != nil {
+	topicConfigs, err := inputAdminClient.DescribeTopicConfigs(ctx, topic)
+	if err != nil {
+		return fmt.Errorf("failed to fetch configs for topic %q from source broker: %s", topic, err)
+	}
+
+	destinationConfigs := make(map[string]*string)
+	topicConfigs.On(topic, func(rc *kadm.ResourceConfig) error {
+		for _, c := range rc.Configs {
+			// Source: https://docs.redpanda.com/current/reference/properties/topic-properties/
+			if c.Key == "cleanup.policy" ||
+				c.Key == "flush.bytes" ||
+				c.Key == "flush.ms" ||
+				c.Key == "initial.retention.local.target.ms" ||
+				c.Key == "retention.bytes" ||
+				c.Key == "retention.ms" ||
+				c.Key == "segment.ms" ||
+				c.Key == "segment.bytes" ||
+				c.Key == "compression.type" ||
+				c.Key == "message.timestamp.type" ||
+				c.Key == "max.message.bytes" ||
+				c.Key == "replication.factor" ||
+				c.Key == "write.caching" ||
+				c.Key == "redpanda.iceberg.mode" ||
+				strings.HasPrefix(c.Key, "redpanda.remote.") {
```

Comment:
Is this correct? Do we have to first query the destination server settings to ensure that we're not overwriting a custom setting?